Leader Key/UserState.swift

```diff
@@ -8,6 +8,7 @@ final class UserState: ObservableObject {
   @Published var display: String?
   @Published var currentGroup: Group?
   @Published var isShowingRefreshState: Bool
+  @Published var navigationPath: [Group]?
```

Comment:
```suggestion
  @Published var navigationPath: [Group] = []
```

This is a great addition.

Let's make it always be an array? Empty at first, empty on clear? I think we can skip a bunch of nil checks if so. Makes sense conceptually as well that the _path is empty_ when we haven't navigated anywhere yet.